iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/module/CitationsActionBuilderModuleUtils.java

```diff
@@ -1,56 +1,132 @@
 package eu.dnetlib.iis.wf.export.actionmanager.module;
 
-import java.util.ArrayList;
-import java.util.Map.Entry;
-
 import eu.dnetlib.iis.common.InfoSpaceConstants;
 import eu.dnetlib.iis.common.citations.schemas.CitationEntry;
 import eu.dnetlib.iis.common.model.extrainfo.ExtraInfoConstants;
 import eu.dnetlib.iis.common.model.extrainfo.citations.BlobCitationEntry;
 import eu.dnetlib.iis.common.model.extrainfo.citations.TypedId;
 import eu.dnetlib.iis.export.schemas.Citations;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
 /**
  * {@link Citations} action builder module utilities.
- * @author mhorst
  *
+ * @author mhorst
  */
 public final class CitationsActionBuilderModuleUtils {
-    
+
     // ------------------------- CONSTRUCTORS ----------------------------
-    
-    private CitationsActionBuilderModuleUtils() {}
+
+    private CitationsActionBuilderModuleUtils() {
+    }
 
     // ------------------------- LOGIC -----------------------------------
-    
-    
+
     /**
      * Creates {@link BlobCitationEntry} from {@link CitationEntry}.
-     * Translates confirence level into trust level applying confidenceToTrustLevelFactor.
+     * Translates confidence level into trust level applying confidenceToTrustLevelFactor.
+     */
+    public static BlobCitationEntry build(CitationEntry citationEntry) {
+        return build(citationEntry, PositionBuilder::build, RawTextBuilder::build, IdentifiersBuilder::build);
+    }
+
+    public static BlobCitationEntry build(CitationEntry citationEntry,
+                                          Function<CitationEntry, Integer> positionFn,
+                                          Function<CitationEntry, String> rawTextFn,
+                                          Function<CitationEntry, List<TypedId>> identifiersFn) {
+        BlobCitationEntry blobCitationEntry = new BlobCitationEntry();
+        blobCitationEntry.setPosition(positionFn.apply(citationEntry));
+        blobCitationEntry.setRawText(rawTextFn.apply(citationEntry));
+        blobCitationEntry.setIdentifiers(identifiersFn.apply(citationEntry));
+        return blobCitationEntry;
+    }
+
+    /**
+     * Build position value of blob citation entry from citation entry.
+     */
+    public static class PositionBuilder {
+        public static Integer build(CitationEntry citationEntry) {
+            return citationEntry.getPosition();
```

Comment:
Do we need a whole function to cover this simple one-liner? I will make a generic comment when summarizing the code review.

Reply:
The reason behind this was to separate the code and match it to other builders in this class and make tests simpler. This particular code is simple but others are not that simple but to align the code I've also applied this build pattern here. But as discussed offline I'm going to apply this pattern carefully in the future.